### PR TITLE
Add MySQLi support

### PR DESCRIPTION
**Pull request by deangray82**
kiddivouchers/wider-wallet#2001

To test change the config/database type in Wider Wallet to `mysqli`.

### CODE REVIEW

File: classes/kohana/database/mysqli.php [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+			'point'                     => array('type' => 'string', 'binary' => TRUE),
+			'real unsigned'             => array('type' => 'float', 'min' => '0'),
+			'set'                       => array('type' => 'string'),
+			'smallint unsigned'         => array('type' => 'int', 'min' => '0', 'max' => '65535'),
+			'text'                      => array('type' => 'string', 'character_maximum_length' => '65535'),
+			'tinyblob'                  => array('type' => 'string', 'binary' => TRUE, 'character_maximum_length' => '255'),
+			'tinyint'                   => array('type' => 'int', 'min' => '-128', 'max' => '127'),
+			'tinyint unsigned'          => array('type' => 'int', 'min' => '0', 'max' => '255'),
+			'tinytext'                  => array('type' => 'string', 'character_maximum_length' => '255'),
+			'year'                      => array('type' => 'string'),
```
Comment (Jamesking56) [inline classes/kohana/database/mysqli.php:242]: Year as a string?

Comment (deangray82) [reply classes/kohana/database/mysqli.php:242]: I copied it from here. https://github.com/kiddivouchers/kohana-database/blob/3beab3507c18a2bcc28cc0689c96ae470722fdcc/classes/kohana/database/mysql.php#L260

Comment (Jamesking56) [reply classes/kohana/database/mysqli.php:242]: Hmm weird

Comment (cs278) [reply classes/kohana/database/mysqli.php:242]: It'll be to handle year 0001, I'd imagine.

File: classes/kohana/database/mysqli.php [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+			Database_MySQLi::$_set_names = ! function_exists('mysqli_set_charset');
+		}
+
+		// Extract the connection parameters, adding required variabels
+		extract($this->_config['connection'] + array(
+			'database'   => '',
+			'hostname'   => '',
+			'username'   => '',
+			'password'   => '',
+            'socket'   => '',
```
Comment (cs278) [inline classes/kohana/database/mysqli.php]: Indents

File: classes/kohana/database/mysqli.php [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+			'port'     => 3306,
+			'ssl'      => NULL,
+		));
+
+        // Prevent this information from showing up in traces
+        unset($this->_config['connection']['username'], $this->_config['connection']['password']);
+
+        try
+        {
+            if(is_array($ssl))
```
Comment (cs278) [inline classes/kohana/database/mysqli.php]: White space is all over the shop here

File: classes/kohana/database/mysqli.php [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+        catch (Exception $e)
+        {
+            // No connection exists
+            $this->_connection = NULL;
+
+            throw new Database_Exception(':error', [':error' => $e->getMessage()], $e->getCode());
+        }
+
+
+		// \xFF is a better delimiter, but the PHP driver uses underscore
```
Comment (cs278) [inline classes/kohana/database/mysqli.php:79]: Indent is off

**Review by cs278 — CHANGES_REQUESTED** [review]
Only thing I can see is whitespace issues, add a `.editorconfig`.

**Comment by deangray82** [comment]
> Only thing I can see is whitespace issues, add a .editorconfig.

@cs278 Updated.

**Review by cs278 — APPROVED** [review]

**Pull request merged**